scenes/menus/storybook/components/quest.gd

```diff
@@ -19,12 +18,30 @@ extends Resource
 ## Leave blank if not needed.
 @export var affiliation: String
 
-@export_group("Advanced")
-
 ## The path to the first scene of the quest.
-## The path can be absolute (beginning with [code]res://[/code])
-## or relative to the folder where this quest resource is saved.
-@export_file("*.tscn") var first_scene: String = "0_intro_template/intro_template.tscn"
+@export_file("*.tscn") var first_scene: String
+
+@export_group("Animation")
+
+## An optional sprite frame library to show in the storybook page for this quest.
+## This could be the main character, an NPC, or an important item in the quest.
+@export var sprite_frames: SpriteFrames:
+	set(new_value):
+		sprite_frames = new_value
+		notify_property_list_changed()
+
+## The animation in [member sprite_frames] to display. This should typically be a looping animation.
+@export var animation_name: StringName = &""
+
+
+func _validate_property(property: Dictionary) -> void:
+	match property["name"]:
+		"animation_name":
+			if sprite_frames:
+				property.hint = PROPERTY_HINT_ENUM
+				property.hint_string = ",".join(sprite_frames.get_animation_names())
+			else:
+				property.usage |= PROPERTY_USAGE_READ_ONLY
```

Comment:
This is of course adapted from `teleporter.gd` – it really makes it nicer to use. I was surprised that Resource subclasses could be tool scripts but I guess why not!